File: leetcode-go/Q051-Q100/Q075_Sort_Colors/Q075_Sort_Colors.go
package Q075_Sort_Colors

func sortColors(nums []int) {
	var n0, n1, n2 int
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			nums[n2] = 2
			n2++
			nums[n1] = 1
			n1++
			nums[n0] = 0
			n0++
		} else if nums[i] == 1 {
			nums[n2] = 2
			n2++
			nums[n1] = 1
			n1++
		} else if nums[i] == 2 {
			nums[n2] = 2
			n2++
		}
	}
}

File: src/facebook/Q075_Sort_Colors.java
package facebook;

public class Q075_Sort_Colors {

    public void sortColors(int[] nums) {
        int l = 0, r = nums.length - 1;
        for (int i = 0; i <= r; i++) {
            if (nums[i] == 0) {
                swap(nums, i, l++);
            } else if (nums[i] == 2) {
                swap(nums, i--, r--);
            }
        }
    }

    private void swap(int[] nums, int i, int j) {
        int tmp = nums[i];
        nums[i] = nums[j];
        nums[j] = tmp;
    }

}
